legion/engine/core/containers/sparse_set.hpp

```diff
@@ -198,6 +199,17 @@ namespace legion::core
             return false;
         }
 
+        /**@brief Finds the index of a value in the dense array
+         * @param val Value to find.
+         * @returns std::optional to the index, otherwise nullopt.
+         */
+        L_NODISCARD std::optional<size_type> index_of(const_reference val)
+        {
+            if (contains(val))
+                return m_sparse[val];
+            return std::nullopt;
+        }
```

Comment:
generally I wouldn't use optional for this, but instead the standard defined invalid index of `static_cast<size_type>(-1)` aka `std::string::npos` aka `std::numeric_limits<size_type>::max()`
in which case defining a `sparse_set::npos` or something similar might also be a good idea